platform/representation/src/gestures/DropGesture.js

```diff
@@ -90,25 +107,61 @@ define(
                 }
             }
 
+            function createVirtualPanel(base, overlayId){
+
+                var typeKey = 'telemetry.panel',
+                    type = typeService.getType(typeKey),
+                    model = type.getInitialModel(),
+                    id = uuid(),
+                    newPanel = undefined;
+
+                model.type = typeKey;
+                model.name = 'New telemetry panel';
+                newPanel = instantiate(model, id);
+
+                [base.getId(), overlayId].forEach(function(id){
+                    newPanel.getCapability('composition').add(id)
+                });
+
+                newPanel.getCapability('location').setPrimaryLocation(base.getCapability('location').getContextualLocation());
+
+                //ObjectService is wrapped by a decorator which is obscuring
+                // the newObject method.
+                var virtualPanel = new EditableDomainObject(newPanel, $q);
+                virtualPanel.setOriginalObject(base);
+                return virtualPanel;
+
+            }
+
             function drop(e) {
                 var event = (e || {}).originalEvent || e,
                     id = event.dataTransfer.getData(GestureConstants.MCT_DRAG_TYPE),
-                    domainObjectType = domainObject.getModel().type;
-                
+                    domainObjectType = editableDomainObject.getModel().type,
+                    virtualObj;
+
                 // If currently in edit mode allow drag and drop gestures to the
                 // domain object. An exception to this is folders which have drop
                 // gestures in browse mode.
-                if (domainObjectType === 'folder' || domainObject.hasCapability('editor')) {
+                //if (domainObjectType === 'folder' || domainObject.hasCapability('editor')) {
                 
                     // Handle the drop; add the dropped identifier to the
                     // destination domain object's composition, and persist
                     // the change.
                     if (id) {
-                        $q.when(action && action.perform()).then(function (result) {
```

Comment:
Actions could return the new EditableDomainObject which could be used directly for navigation. In the case of telemetry-on-telemetry drops, would be an EditableDomainObject wrapping a new 'virtual' panel, otherwise just an EditableDomainObject wrapping the domain object being edited.